src/middleware.js

```diff
@@ -118,7 +118,14 @@ function apiMiddleware({ getState }) {
     }
 
     // We can now dispatch the request FSA
-    next(await actionWith(requestType, [action, getState()]));
+    if (
+      typeof requestType.payload === 'function' ||
+      typeof requestType.meta === 'function'
+    ) {
+      next(await actionWith(requestType, [action, getState()]));
+    } else {
+      next(requestType);
```

Comment:
Is it possible/worth it to add tests that verify this does happen synchronously, as expected?

Reply:
I think it's neither worth nor possible to test that this happens synchronously. Provided there is no mistake in `if` condition, and taking into account there is no `await` in `else` branch, this should work as expected. I have added 2 tests that cover first branch of this `if` statement, and they ensure that `meta` or `payload` functions are called, that they are called with correct arguments, and that their result is inserted to `meta` or `payload` property of request FSA. A lot of existing tests cover the second branch.